sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/state/SymmetricHashJoinStateManager.scala

```diff
@@ -274,6 +274,9 @@ class SymmetricHashJoinStateManager(
           if (valuePairAtMaxIndex != null) {
             keyWithIndexToValue.put(currentKey, index, valuePairAtMaxIndex.value,
               valuePairAtMaxIndex.matched)
+          } else {
+            logWarning(s"`keyWithIndexToValue` returns a null value for index ${numValues - 1} " +
```

Comment:
The reason I suggested to log a warning message here is to help debugging if the problem arises.

Suppose we see the log message and would like to understand what's happening. Does it give enough information, with other logs? Can we determine the current key from the log context?

Reply:
Hm, as currentKey is an UnsafeRow, I guess it is hard to get understanding from its string output?

Reply:
I can log the currentKey if you think it is important to have.

Reply:
Sigh that's the problem... We won't be able to reproduce a key row from string output.

Probably we need to think more on this. I don't like to swallow the problem we can catch in prior, but I don't also like the way we log cryptic message to end user, and when they come up with the log message we say "Sorry we don't have enough information from the log.". Cryptic log message might be acceptable, but at least we should be able to provide the way how to investigate further.

The next possible way may be... having state store reader and scan all state store to find null in value? If we are OK with this one, then it might still make sense to leave this log message.

Less cryptic, or guide to report to Spark community might be better though. Do we have some way to log an internal error so that community can report it?

Reply:
Less cryptic approach might be to project unsafe row to internal row. Although it needs extra projection, but I think this is less likely case to happen. So maybe we don't need to worry too much about performance.

WDYT?

Reply:
Oh that's a brilliant idea. Just to make sure we don't miss anything, how it will be printed out? Probably testing with simple but composite row (like row having String, Int, Timestamp, etc.) would be good.

Reply:
Okay, let me do it and add some tests.